S3FileRenamer.py

```diff
@@ -23,18 +21,33 @@ def contains_bad_characters(string):
     else:
         return False, ""
 
-
 def perform_rename(old_name, root_path, found_bad_characters, rename_do):
```

Comment:
can you change rename_do to do_rename?

Reply:
I guess you could, but then it would be the same variable name as the one outside the function.
I don't like variables having the same name, ever since my high school programming teacher said that one of the Voyager probes was lost because someone used a non-exclusive variable name and it conflicted with another variable with the same name... :)

Reply:
Which begs the question: why are you using the global variable `bad_characters` when it is tightly coupled to the `contains_bad_characters()` function?  You should either define it in there, or use a literal instead (or pass it as a parameter).

Reply:
Thanks @pmather!